### PR TITLE
fix: #4196 suggestion panel not handling collision detection

### PR DESCRIPTION
**Pull request by thejus557**
fix: #4196  suggestion panel not handling collision detection

https://github.com/primefaces/primereact/assets/38601497/a475558f-cd85-45e1-a7a7-9f593b3f142a

### CODE REVIEW

**Comment by thejus557** [comment]
@melloware  Please take a look at this pr.

File: components/lib/mention/Mention.js [hunk 1]
```diff
@@ -94,9 +94,20 @@ export const Mention = React.memo(
...
             overlayRef.current.style.transformOrigin = 'top';
-            overlayRef.current.style.left = `calc(${position.left}px + 1rem)`;
+
+            const isTouchingRight = left >= viewportWidth - inputRefClientRect.right;
+
+            if (isTouchingRight) {
+                overlayRef.current.style.left = `calc(${viewportWidth - inputRefClientRect.right}px - 1rem)`;
+            } else {
+                overlayRef.current.style.left = `calc(${position.left}px + 1rem)`;
+            }
```
Comment (melloware) [inline components/lib/mention/Mention.js:109]: I wonder if its something that can be fixed in `DomHandler.flipfitCollision` more generically to be re-usable?

Comment (thejus557) [reply components/lib/mention/Mention.js:109]: @melloware  Do you think the below code should work ? 

```
        const alignOverlay = () => {
            DomHandler.flipfitCollision(inputRef.current, overlayRef.current, '', 'left bottom');
        };
```
right now its taking top:0, left:0 with above. Am i missing anything. Please let me know.

Screenshot 2023-08-09 at 1 55 08 AM

Comment (melloware) [reply components/lib/mention/Mention.js:109]: I mean it should work but I tink it wasn't working when I tried it before?  Does it seem to be working now?

Comment (suryateja-kandukuru) [reply components/lib/mention/Mention.js:109]: flipfitCollision is working but its keeping the position top:0 and left:0 Let me look into this deeper tomorrow.

**Comment by lagroms** [comment]
any news for this feature ? It's blocking me to implement it

**Pull request closed without merging**